src/filesystem/impls/filer/BlobUtils.js

```diff
@@ -0,0 +1,93 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, URL */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Handlers = require("filesystem/impls/lib/handlers");
+    var Log = require("filesystem/impls/lib/log");
+
+    // BlobUtils provides an opportunistic cache for BLOB Object URLs
+    // which can be looked-up synchronously.
+
+    var Filer = require("thirdparty/filer/dist/filer");
+    var Path  = Filer.Path;
+
+    // 2-way cache for blob URL to path for looking up either way:
+    // * urls - paths keyed on blobUrls
+    // * blobs - blobUrls keyed on paths
+    var urls  = {};
+    var blobs = {};
+
+    // Generate a BLOB URL for the given filename and cache it
+    function cache(filename, callback) {
+        filename = Path.normalize(filename);
+
+        Handlers.handleFile(Filer.fs(), filename, function(url) {
```

Comment:
Acutally, why not just let the `Handlers` module deal with getting the `fs` instance itself vs. having to pass it into the `handleFile` function?